src/types/utils.d.ts

```diff
@@ -141,3 +141,5 @@ type PropertyTypeOrFallback<T, Key extends string, Fallback> = Key extends keyof
  * Makes the given keys `K` of the type `T` required
  */
 type RequiredProps<T, K extends keyof T> = Required<Pick<T, K>> & Omit<T, K>;
+
+type StringNumber<S extends string> = S extends `${number}` ? S : never;
```

Comment:
I'd call this `NumberToString`, on the first usage I saw, I sort thought it was the inverse; a string to a number.